sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/TargetScaleThrottler.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.WebJobs.EventHubs.Listeners;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.EventHubs.Listeners
+{
+    internal class TargetScaleThrottler
+    {
+        // Throttle scale in requests if last scale out time was within ThrottleScaleDownIntervalInSeconds.
+        private const int ThrottleScaleDownIntervalInSeconds = 180;
+
+        // Keep metrics and target scale resut history for ExpirationThresholdInseconds
+        private const int ExpirationThresholdInseconds = 60;
+
+        private const int ScaleUpFactor = 10;
+
+        private readonly List<EventHubTragetScalerReult> _history = new List<EventHubTragetScalerReult>();
+        private readonly ILogger _logger;
+        private object lockObject = new object();
+
+        private TargetScalerResult _lastTargetScalerResult = new TargetScalerResult();
+        private DateTime _lastScaleUpTime = DateTime.MinValue;
+
+        public TargetScaleThrottler(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        // for tests
+        public TargetScaleThrottler(DateTime lastScaleUpTime, TargetScalerResult lastTargetScalerResult, ILogger logger)
+        {
+            _lastScaleUpTime = lastScaleUpTime;
+            _lastTargetScalerResult = lastTargetScalerResult;
+            _logger = logger;
+        }
+
+        public TargetScalerResult ThrottleIfNeeded(TargetScalerResult result, EventHubsTriggerMetrics metric, DateTime executionTime)
+        {
+            try
+            {
+                lock (lockObject)
+                {
+                    // Removing expired items from history. We want to keep the history item around the threshold so adding 2 secs.
```

Comment:
I'm confused by this since the constant is only used right here. What difference does it make to add 2 seconds if we are defining it arbitrarily and it is only used here. Or is this set to 60s somewhere else?

Reply:
yeah, it used only in one place. Set it to `ExpirationThresholdInSeconds = 65`.